tests/lib/acpi/src/main.c

```diff
@@ -17,14 +17,43 @@ ZTEST(acpi, test_mcfg_table)
 	zassert_not_null(mcfg, "Failed to get MCFG table");
 }
 
-ZTEST(acpi, test_irq_routing_table)
+#if ACPI_DT_HAS_HID(DT_NODELABEL(pcie0))
+ZTEST(acpi, test_dev_enum)
 {
```

Comment:
Please don't use this style to skip tests. Instead, use `ztest_test_skip()`, e.g.:
```c
	if (!ACPI_DT_HAS_HID(DT_NODELABEL(pcie0))) {
		test_test_skip();
		return;
	}
```

The reasoning for this is explained here: https://docs.zephyrproject.org/latest/develop/test/ztest.html#skipping-tests

Reply:
There is also Z_TEST_SKIP_IFNDEF(config)

Reply:
defined as  `COND_CODE_1(config, (), (ztest_test_skip()))`

Reply:
@jhedberg ztest_test_skip() should not return

Reply:
Btw, I can see that there are many places under `tests/` with explicit `return;` statements after `z_test_skip();`, but I guess that's just redundant/dead code then.

Reply:
https://github.com/zephyrproject-rtos/zephyr/blob/a3ff19a39eb18bc10ae38698f5c91ad50c5748d7/subsys/testsuite/ztest/src/ztest.c#L334-L346

Reply:
The test might not compile using this ztest_test_skip() , then `ifdef` should be OK.

Reply:
> The test might not compile using this ztest_test_skip() , then `ifdef` should be OK.

No, even in that case it's not ok (you can see the reasoning in the Zephyr documentation that I linked). If the code doesn't build, then you should still use `ztest_test_skip()` in a separate pre-processor branch. You could do that by having the pre-processor condition inside the test function body, together with an `#else`, or then by having two variants of the same test function (one of which would simply contain the skip directive). The point with all this is that all functions are always visible in the test statistics & output, and we can see whenever some tests get skipped, instead of them just silently being compiled out without any trace.